thriftbp/server.go

```diff
@@ -72,7 +88,7 @@ func NewServer(cfg ServerConfig) (*thrift.TSimpleServer, error) {
 	var transport *thrift.TServerSocket
 	if cfg.Socket == nil {
 		var err error
-		transport, err = thrift.NewTServerSocket(cfg.Addr)
+		transport, err = thrift.NewTServerSocketTimeout(cfg.Addr, cfg.ThriftSocketTimeout)
```

Comment:
I assume this doesn't set a timeout if it's zero?

I _might_ do an explicit

```
if cfg.ThriftSocketTimeout > 0 && cfg.Socket == nil { ...
  // call NewTServerSocketTimeout
  cfg.Logger = ...
} else if cfg.Socket == nil {
  // original code
}
```

This keeps the special / experimental codepath easy to see and keeps all of the repercussions of setting it in a single narrow branch

Reply:
calling `NewTServerSocketTimeout` with 0 timeout is the same as no timeout. but calling `NewTServerSocketTimeout` with negative timeout could have undefined behavior. so +1 to add an explicit check.